lib/client.js

```diff
@@ -597,6 +597,21 @@ MqttClient.prototype.end = function (force, cb) {
 }
 
 /**
+ * removeOutgoingStore - remove a message in outgoing store
+ *
+ * @param {Number} mid - messageId to remove message
+ * @returns {MqttClient} this - for chaining
+ * @api public
+ *
+ * @example client.removeOutgoingStore(client.getLastMessageId());
+ */
+MqttClient.prototype.removeOutgoingStore = function (mid) {
+  delete this.outgoing[mid]
```

Comment:
I agree. It is better to inform the client the result. I updated the code, test, and document.